src/dawn/IIR/MultiStage.cpp

```diff
@@ -129,6 +129,30 @@ iir::Cache& MultiStage::setCache(iir::Cache::CacheTypeKind type, iir::Cache::Cac
       .first->second;
 }
 
+Interval::IntervalLevel MultiStage::lastLevelComputed(const int accessID) const {
+  Interval::IntervalLevel level;
+  bool init = false;
+  for(const auto& doMethod : iterateIIROver<DoMethod>(*this)) {
+    if(!doMethod->getFields().count(accessID)) {
+      continue;
+    }
+    const auto& interval = doMethod->getInterval();
+
+    if(loopOrder_ == LoopOrderKind::LK_Backward) {
+      if(interval.bound(Interval::Bound::lower) < level.bound() || !init) {
+        level = interval.lowerIntervalLevel();
+        init = true;
```

Comment:
why not just return early if you only want to check once? And is this the intended behavior?

Reply:
this pattern is like computing a min, you might enter multiple times in the level =, which gets updated , so the early return would not work ehre